### DISCUSSION TITLE
AWS IAm role support for S3 Access ?

### DISCUSSION THREAD

**Discussion by scott-dunt** [outline/outline#2386] [Ideas]
When running applications on 'EC2' hosts in Amazon Web Services, AWS, we prefer to use 'IAM Roles' to enable access to AWS resources like S3 buckets. This way we don't have security ID's and password - keys stored in files on the hosts, the access is allowed and assigned to the HOST system internally by AWs: https://docs.aws.amazon.com/IAM/latest/UserGuide/id_roles_use_switch-role-ec2.html

My testing seems to indicate that the Outline app does not work with IAM roles.  I have tried using no and blank credentials in the .env file and neither works:

AWS_ACCESS_KEY_ID=
AWS_SECRET_ACCESS_KEY=

Any chance to add this functionality?
https://docs.aws.amazon.com/AWSEC2/latest/UserGuide/iam-roles-for-amazon-ec2.html#instance-metadata-security-credentials

**Comment by tommoor** [comment]
Open to this as a contribution – the only interface with S3 is through their SDK so I'm kind of surprised it doesn't work out of the box to be honest, it's likely to be a small change.

**Comment by l2dy** [comment]
Implemented in the main branch, please try it out.
